analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -225,11 +227,13 @@ class NuGetSupport(serviceIndexUrls: List<String> = listOf(DEFAULT_SERVICE_INDEX
                         all.details.dependencyGroups.flatMapTo(mutableSetOf()) { it.dependencies }
 
                     buildDependencyTree(
-                        referredDependencies.map {
-                            // Simply take the minimum of the Ivy-style version range.
-                            val version = it.range.removePrefix("[").substringBefore(",").trim()
+                        referredDependencies.map { dependency ->
+                            // Resolve to the highest existent version that matches the version description as specified
+                            // at https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges.
+                            val version = dependency.range.trim { it.isWhitespace() || it in VERSION_RANGE_CHARS }
```

Comment:
Hint: above link says "NuGet 2.8.x and earlier chooses the latest available package version when resolving a dependency, whereas NuGet 3.x and later chooses the lowest package version. Specifying a version or version range avoids this uncertainty." which I found quite surprising.

Reply:
Luckily, that behavior only kicks in "Without a version or version range" specified at all.